### ISSUE TITLE
Feedback: provide choice of render engine

### ISSUE THREAD

**Issue by dotNomad** [posit-dev/publisher#2059]
from: https://positpbc.slack.com/archives/C06B0UZKT34/p1722397011679269

I have a jupyter notebook that I want to render as a quarto doc. I would like an option in the publisher when creating the deployment to choose if quarto or jupyter notebook is used to render.

-----

From Jon:
>I looked for an extant ticket already, but couldn't find one. So I've created https://github.com/rstudio/connect/issues/27842 for (possibly) changing this default in Connect. Like I mention in the issue, this is really an issue about Connect's default. IMHO the publisher shouldn't override the default, if we think the default is wrong we should change it on Connect.

**Comment by dotNomad** [comment]
We should make notebooks on Connect by default render, or use the default based on the frontmatter. Ideally this is not behavior we should change in the Publisher.

Since https://github.com/rstudio/connect/issues/27842 is up in Connect closing.

**Comment by SamEdwardes** [comment]
I think this suggestion makes sense. Connect can decide how to render based on the front matter.

How would the logic work? For example if there is no front matter do you default to rendering as a jupyter notebook? But if there is front matter you render based on that?

**Comment by dotNomad** [comment]
> For example if there is no front matter do you default to rendering as a jupyter notebook?

Fantastic question. I mentioned we would talk about this again in our sync so I'll re-open and throw on our Triage board to get that discussion going.

**Comment by mmarchetti** [comment]
Triage notes:
* The request in this issue is to make `.ipynb` files deployable as Quarto docs (we removed this at one point)
* We'll need to document the behavior of notebooks deployed in ``notebook mode (code cells are executed) vs. in Quarto mode (they aren't executed unless you set the `execute` flag in the Quarto YAML.
* When we have commented configurations, we should add a note about this in the configuration file.
